src/lib.rs

```diff
@@ -19,33 +19,25 @@ fn parse_quoted(input: &str) -> IResult<&str, String> {
     Ok(res)
 }
 
-fn and_parser(i: &str) -> IResult<&str, &str> {
-    tag("&&")(i)
-}
-fn or_parser(i: &str) -> IResult<&str, &str> {
-    tag("||")(i)
-}
-fn and_or_choice(i: &str) -> IResult<&str, &str> {
-    alt((and_parser, or_parser))(i)
-}
 
-fn base_parser(i: &str) -> IResult<&str, BaseOutput, Error<&str>> {
-    let f =  tag("foo");
-    let eq_op  =  tag("==");
+fn base_parser<'a>() -> impl FnMut(&'a str) -> IResult<&'a str, BaseOutput, Error<&'a str>> {
```

Comment:
`<'a>() -> impl FnMut(&'a str) -> IResult<&'a str, BaseOutput, Error<&'a str>>`
this means that base_parser returns a function which returns the parser? 
I like it!!

Reply:
not sure what you meant,
if you meant that `base_parser` **is** a function that returns the parser, so yes.
because parser could be a function(A parser is actually a trait that auto-implemented for function with the correct syntax) 
`base_parser` is a function that will return a function, this function is the parser.